refinery/ui/source/js/user-file-browser/services/factory.js

```diff
@@ -124,9 +124,11 @@
             });
           }
         });
-        filters[filterName] = {
-          facetObj: facetCounts
-        };
+        if (facetCounts.length > 0) {
```

Comment:
Design choice: once a user starts filtering, they would not see the available fields.  Also, I believe solr has a min count filter, so you can add it to the query instead of filtering client side. https://wiki.apache.org/solr/SimpleFacetParameters#facet.mincount
![short-zero](https://user-images.githubusercontent.com/9956764/27560702-2f24018e-5a93-11e7-9a18-0e57e909afec.gif)